packages/devtools_app/test/shared/auto_dispose_mixin_test.dart

```diff
@@ -105,19 +103,131 @@ void main() {
       disposer.addAutoDisposeListener(notifier, () {
         values.add(notifier.value);
       });
-      // ignore: invalid_use_of_protected_member
       expect(notifier.hasListeners, isTrue);
       notifier.value = 19;
       expect(values.length, equals(3));
       expect(values.last, equals(19));
       disposer.cancelListeners();
 
-      // ignore: invalid_use_of_protected_member
       expect(notifier.hasListeners, isFalse);
       notifier.value = 21;
       expect(values.length, equals(3));
       expect(values.last, equals(19));
     });
+
+    group('callOnceWhenReady', () {
+      for (bool readyWhen in [false, true]) {
+        group('readyWhen=$readyWhen', () {
+          testWidgets('triggers callback and cancels listeners when ready ',
+              (WidgetTester tester) async {
+            final disposer = Disposer();
+            final trigger = ValueNotifier<bool?>(!readyWhen);
+            final callbackEntries = <int>[];
+            int counter = 0;
```

Comment:
we never actually check the value of counter. Either only check the length of callbackEntries or only check the value of counter. If we aren't ever checking the actual values in callbackEntries, then I'd recommend we just check the value of counter